test/unit/test.sign.js

```diff
@@ -894,6 +898,48 @@ describe('sign', function() {
     }).catch(done);
   });
 
+  it('removes created XPI after signing', function(done) {
```

Comment:
The jpm sign command isn't responsible for creating unsigned files. You can use `jpm xpi` for that. I don't think this ever should have kept the XPI around. All the other commands that create temporary XPIs [remove them](https://github.com/mozilla-jetpack/jpm/blob/master/lib/task.js#L45) afterwards. I didn't realize that because I was mainly modeling `jpm xpi`.